containers/ecr-viewer/src/app/api/save-fhir-data/save-fhir-data-service.ts

```diff
@@ -171,355 +173,208 @@ const saveFhirMetadata = async (
-              lab.test_result_quantitative,
-            )
-            .input("test_result_units", sql.VarChar(50), lab.test_result_units)
-            .input("test_result_code", sql.VarChar(50), lab.test_result_code)
-            .input(
-              "test_result_code_display",
-              sql.VarChar(255),
-              lab.test_result_code_display,
-            )
-            .input(
-              "test_result_code_system",
-              sql.VarChar(50),
-              lab.test_result_code_system,
-            )
-            .input(
-              "test_result_interpretation",
-              sql.VarChar(255),
-              lab.test_result_interpretation,
-            )
-            .input(
-              "test_result_interpretation_code",
-              sql.VarChar(50),
-              lab.test_result_interpretation_code,
-            )
-            .input(
-              "test_result_interpretation_system",
-              sql.VarChar(255),
-              lab.test_result_interpretation_system,
-            )
-            .input(
-              "test_result_ref_range_low_value",
-              sql.Float,
-              lab.test_result_ref_range_low,
-            )
-            .input(
-              "test_result_ref_range_low_units",
-              sql.VarChar(50),
-              lab.test_result_ref_range_low_units,
-            )
-            .input(
-              "test_result_ref_range_high_value",
-              sql.Float,
-              lab.test_result_ref_range_high,
-            )
-            .input(
-              "test_result_ref_range_high_units",
-              sql.VarChar(50),
-              lab.test_result_ref_range_high_units,
-            )
-            .input("specimen_type", sql.VarChar(255), lab.specimen_type)
-            .input(
-              "specimen_collection_date",
-              sql.DateTime,
-              lab.specimen_collection_date,
-            )
-            .input("performing_lab", sql.VarChar(255), lab.performing_lab)
-            .query(
-              "INSERT INTO ecr_viewer.ecr_labs VALUES (@UUID, @eICR_ID, @test_type, @test_type_code, @test_type_system, @test_result_qualitative, @test_result_quantitative, @test_result_units, @test_result_code, @test_result_code_display, @test_result_code_system, @test_result_interpretation, @test_result_interpretation_code, @test_result_interpretation_system, @test_result_ref_range_low_value, @test_result_ref_range_low_units, @test_result_ref_range_high_value, @test_result_ref_range_high_units, @specimen_type, @specimen_collection_date, @performing_lab)",
-            );
+  try {
+    await getDb<Extended>()
+      .transaction()
+      .execute(async (trx) => {
+        await trx
+          .insertInto("ecr_data")
+          .values({
+            eicr_id: ecrId,
+            set_id: metadata.eicr_set_id,
+            last_name: metadata.last_name,
+            first_name: metadata.first_name,
+            birth_date: metadata.birth_date,
```

Comment:
do we need to wrap this with `asDate` too?

Reply:
no, because per the fhir spec it always comes in a known good format and is a date, not a date time